pkg/volume/azure_file/azure_file.go

```diff
@@ -104,6 +105,13 @@ func (plugin *azureFilePlugin) newMounterInternal(spec *volume.Spec, pod *v1.Pod
 	if err != nil {
 		return nil, err
 	}
+	nameSpace := pod.Namespace
+	if spec.PersistentVolume != nil {
+		ns, err := getSecretNamespace(plugin.host.GetKubeClient(), spec.PersistentVolume)
```

Comment:
Mounters run on the nodes, and have to be authorized to obtain secrets needed for mounting. This is done by the node authorizer based on https://github.com/kubernetes/kubernetes/blob/master/pkg/api/persistentvolume/util.go#L33

To know which secrets a PV grants access to, we need structured data in the PV, not ad-hoc storage class parameters.